test/parallel/test-internal-assert.js

```diff
@@ -0,0 +1,15 @@
+// Flags: --expose-internals
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const internalAssert = require('internal/assert');
+
+// Should not throw.
+internalAssert(true);
+internalAssert(true, 'fhqwhgads');
+
+assert.throws(() => { internalAssert(false); }, assert.AssertionError);
+assert.throws(() => { internalAssert(false, 'fhqwhgads'); },
+              { code: 'ERR_ASSERTION', message: 'fhqwhgads' });
```

Comment:
Hint: it is possible to check for the name field as well. That is a good alternative to checking for instanceof `assert.AssertionError` above (but it's off course not the same strict check).